framework/src/meshgenerators/TransfiniteMeshGenerator.C

```diff
@@ -0,0 +1,542 @@
+                                         const Point & P2,
+                                         const std::string & parameter,
+                                         const Point & outward,
+                                         const std::vector<Real> & param_vec)
+{
+  std::vector<Point> edge; // output, to be returned variable
+  edge.resize(param_vec.size());
+
+  std::vector<Real> param_coords;
+  MooseUtils::tokenizeAndConvert(parameter, param_coords, "&&");
+  Point P3;
+  if (param_coords.size() == 1)
+  {
+    P3 = computeMidPoint(P1, P2, param_coords[0], outward);
+  }
+  else
+  {
+    P3 = Point(param_coords[0], param_coords[1], param_coords[2]);
+  }
+
+  Real rad = computeRadius(P1, P2, P3);
+  Point P0 = computeOrigin(P1, P2, P3);
+
+  // need to shift to center of coordinates to find the corresponding radians
+  Point x0 = (P1 - P0);
+  Point x1 = (P2 - P0);
+
+  // The case when the edge spans quadrants 1 and 4 requires special treament
+  // to periodically switch we compute the angle that needs added to one edge
+  // to identify the entire edge span
+  mooseAssert(x0.norm() > 0.0 && x1.norm() > 0.0,
+              "The point provided cannot generate an arc circle on the edge specified");
+
+  Real arclength = std::acos((x0 * x1) / x0.norm() / x1.norm());
+
+  Real a = std::atan2(x0(1), x0(0));
+  Real b = std::atan2(x1(1), x1(0));
+
+  if (a < 0)
+    a = a + 2 * M_PI;
+  if (std::abs(b - a) > M_PI)
+    b = a + arclength;
+
+  auto it = 0;
+  for (auto rx : param_vec)
+  {
+    Real interval = getMapInterval(rx, 0.0, 1.0, a, b);
+    Real x = P0(0) + rad * std::cos(interval);
+    Real y = P0(1) + rad * std::sin(interval);
+    edge[it] = Point(x, y, 0.0);
+    it++;
+  };
+
+  return edge;
+}
+
+Real
+TransfiniteMeshGenerator::getMapInterval(
+    const Real & xab, const Real & a, const Real & b, const Real & c, const Real & d) const
+// this routine maps a point x\in[a, b] to the corresponding point in the interval [c, d]
+{
+  mooseAssert(std::abs(b - a) > 0.0,
+              "The input interval [a, b] is empty, check that a and b are not identical.");
+  Real xcd = c + (d - c) / (b - a) * (xab - a);
+
+  return xcd;
+}
+
+std::vector<Real>
+TransfiniteMeshGenerator::getPointsDistribution(const Real & edge_length,
+                                                const unsigned int & np,
+                                                const Real & bias) const
+{
+  std::vector<Real> param_vec;
+  Real rx = 0.0;
+
+  if (bias != 1.0)
+  {
+    Real step = 0.0;
+    Real factor = edge_length * (1.0 - std::abs(bias)) / (1.0 - std::pow(std::abs(bias), np - 1));
+    param_vec.push_back(rx);
+    for (unsigned int iter = 1; iter < np; iter++)
+    {
+      step = step + factor * std::pow(bias, Real(iter - 1));
+      rx = getMapInterval(step, 0.0, edge_length, 0.0, 1.0);
+      param_vec.push_back(rx);
+    }
+  }
+  else
+  {
+    const Real interval = edge_length / Real(np - 1);
+    Real rx = 0.0;
```

Comment:
This is shadowing the previous declaration.  Not a bug, but it'll start screaming bloody murder if we turn on a few more compiler warnings.